engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1495,6 +1509,11 @@ private void validateThreadLocal(OperatorMeta om) {
       return;
     }
 
+    if (om.getOperator() instanceof Operator.DelayOperator) {
+      String msg = String.format("Locality %s invalid for delay operator %s", Locality.THREAD_LOCAL, om);
+      throw new ValidationException(msg);
```

Comment:
This is because if there is a THREAD_LOCAL for the loop, it would end up being an infinite recursive call within the thread.  However, it's technically possible to have THREAD_LOCAL for part of the cycle.  I have reopened this JIRA: https://issues.apache.org/jira/browse/APEXCORE-262.  We will tackle this in the future.